zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/ProposalRequestProcessor.java

```diff
@@ -89,4 +103,14 @@ public void shutdown() {
         syncProcessor.shutdown();
     }
 
+    private boolean shouldForwardToNextProcessor(Request request) {
+        if (!forwardLearnerRequestsToCommitProcessorDisabled) {
+            return true;
+        }
+        if (request.getOwner() instanceof LearnerHandler) {
```

Comment:
What about a more Object Oriented design like

if (request.getOwner().allowForwardToCommitProcessor(request)).....

Reply:
That seems a little bit overkill to me, the request owner will be the 'me' Object defined in ServerCnxn or LearnerHandler. In order to do this, we need to define an interface which doesn't have other common behavior, which seems not necessary.

Reply:
Makes sense